src/cli/mod.rs

```diff
@@ -0,0 +1,239 @@
+    limit: Option<usize>,
+    // /// Reduction count used for proofs (defaults to 10)
+    // #[clap(long, value_parser)]
+    // rc: Option<usize>,
+
+    // /// Flag to prove the last evaluation
+    // #[arg(long)]
+    // prove: bool,
+}
+
+#[derive(Parser, Debug)]
+struct LoadCli {
+    #[clap(value_parser)]
+    lurk_file: PathBuf,
+
+    #[clap(long, value_parser)]
+    zstore: Option<PathBuf>,
+
+    #[clap(long, value_parser)]
+    limit: Option<usize>,
+    // #[arg(long)]
+    // prove: bool,
+
+    // #[clap(long, value_parser)]
+    // rc: Option<usize>,
+}
+
+impl LoadArgs {
+    pub fn into_cli(self) -> LoadCli {
+        LoadCli {
+            lurk_file: self.lurk_file,
+            zstore: self.zstore,
+            limit: self.limit,
+            // prove: self.prove,
+            // rc: self.rc,
+        }
+    }
+}
+
+#[derive(Args, Debug)]
+struct ReplArgs {
+    /// ZStore to be preloaded before entering the REPL (and loading a file)
+    #[clap(long, value_parser)]
+    zstore: Option<PathBuf>,
+
+    /// Optional file to be loaded before entering the REPL
+    #[clap(long, value_parser)]
+    load: Option<PathBuf>,
+
+    /// Maximum number of iterations allowed (defaults to 100_000_000)
+    #[clap(long, value_parser)]
+    limit: Option<usize>,
+    // /// Reduction count used for proofs (defaults to 10)
+    // #[clap(long, value_parser)]
+    // rc: Option<usize>,
+}
+
+#[derive(Parser, Debug)]
+struct ReplCli {
+    #[clap(long, value_parser)]
+    load: Option<PathBuf>,
+
+    #[clap(long, value_parser)]
+    zstore: Option<PathBuf>,
+
+    #[clap(long, value_parser)]
+    limit: Option<usize>,
+    // #[clap(long, value_parser)]
+    // rc: Option<usize>,
+}
+
+impl ReplArgs {
+    pub fn into_cli(self) -> ReplCli {
+        ReplCli {
+            load: self.load,
+            zstore: self.zstore,
+            limit: self.limit,
+            // rc: self.rc,
+        }
+    }
+}
+
+fn get_field() -> Result<LanguageField> {
+    if let Ok(lurk_field) = std::env::var("LURK_FIELD") {
+        match lurk_field.to_lowercase().as_str() {
+            "bls12-381" => Ok(LanguageField::BLS12_381),
+            "pallas" => Ok(LanguageField::Pallas),
+            "vesta" => Ok(LanguageField::Vesta),
+            _ => bail!("Field not supported: {lurk_field}"),
+        }
+    } else {
+        Ok(DEFAULT_FIELD)
+    }
+}
+
+fn get_store<F: LurkField + for<'a> serde::de::Deserialize<'a>>(
+    zstore: &Option<PathBuf>,
+) -> Store<F> {
+    zstore
+        .as_ref()
+        .and_then(|zstore_path| fs::read(zstore_path).ok())
+        .and_then(|zstore_bytes| ZData::from_bytes(&zstore_bytes).ok())
+        .and_then(|zstore_data| from_z_data(&zstore_data).ok())
+        .map(|zstore: ZStore<F>| zstore.to_store())
+        .tap_none(|| {
+            if zstore.is_some() {
+                eprintln!("Failed to load ZStore. Starting with empty store.")
+            }
+        })
+        .unwrap_or_default()
```

Comment:
The use of `and_then` is appropriate, but you're working with an `Option`, not a `Result`. With a `Result` and `?` you could get a specific message at the point where an operation fails, and it could tell you what the issue is. moreover, this is easy to implement with `anyhow`.

Further, you're probably proceeding in this way because the store path is meant to be read if present and created at that path if "absent", which means that this should succeed in both cases. We can either :
- dissect what the user intended by checking if the filepath is occupied, and trying to read, fallibly, if it is, as suggested above,
- or at least logging what we do, using e.g. `tap` in conjunction with `log::`.

The CLI does set up a logger, let's use it.

Reply:
Done